examples/mnist_gans/callbacks.py

```diff
@@ -0,0 +1,124 @@
+import torch
+import torchvision.utils
+from catalyst.dl import registry
+from catalyst.dl.callbacks.logging import TensorboardLogger
+from catalyst.dl.core import Callback, RunnerState
+from tensorboardX import SummaryWriter
+
+
+@registry.Callback
+class VisualizationCallback(Callback):
+    def __init__(
+        self,
+        input_keys=None,
+        output_keys=None,
+        batch_frequency=25,
+        concat_images=True,
+        max_images=20,
+        n_row=1,
+        denorm="default"
+    ):
+        if input_keys is None:
+            self.input_keys = []
+        elif isinstance(input_keys, str):
+            self.input_keys = [input_keys]
+        elif isinstance(input_keys, (tuple, list)):
+            assert all(isinstance(k, str) for k in input_keys)
+            self.input_keys = list(input_keys)
+        else:
+            raise ValueError(
+                f"Unexpected format of 'input_keys' "
+                f"argument: must be string or list/tuple"
+            )
+
+        if output_keys is None:
+            self.output_keys = []
+        elif isinstance(output_keys, str):
+            self.output_keys = [output_keys]
+        elif isinstance(output_keys, (tuple, list)):
+            assert all(isinstance(k, str) for k in output_keys)
+            self.output_keys = list(output_keys)
+        else:
+            raise ValueError(
+                f"Unexpected format of 'output_keys' "
+                f"argument: must be string or list/tuple"
+            )
+
+        if len(self.input_keys) + len(self.output_keys) == 0:
+            raise ValueError("Useless visualizer: pass at least one image key")
+
+        self.batch_frequency = int(batch_frequency)
+        assert self.batch_frequency > 0
+
+        self.concat_images = concat_images
+        self.max_images = max_images
+        if denorm.lower() == "default":
+            # normalization from [-1, 1] to [0, 1] (the latter is valid for tb)
+            self.denorm = lambda x: x / 2 + .5
+        elif denorm is None or denorm.lower() == "none":
+            self.denorm = lambda x: x
+        else:
+            raise ValueError("unknown denorm fn")
+        self._n_row = n_row
+        self._reset()
+
+    def _reset(self):
+        self._loader_batch_count = 0
+        self._loader_visualized_in_current_epoch = False
+
+    @staticmethod
+    def _get_tensorboard_logger(state: RunnerState) -> SummaryWriter:
+        for logger_name, logger in state.loggers.items():
```

Comment:
you can simply import `state.loggers["tensorboard"]`

Reply:
and what will happen when you rename that logger? =(
what do you think about the new solution? it seems to become even less readable